jwst/refpix/irs2_subtract_reference.py

```diff
@@ -615,96 +628,92 @@ def flag_bad_refpix(datamodel, n_sigma=3.0, flag_only=False, replace_only=False)
 
-                log.debug(f'High diffs={np.sum(high_diffs)}, '
-                          f'high means={np.sum(high_means)}, '
-                          f'high stds={np.sum(high_stds)}')
+                log.debug(
+                    f"High diffs={np.sum(high_diffs)}, "
+                    f"high means={np.sum(high_means)}, "
+                    f"high stds={np.sum(high_stds)}"
+                )
                 int_bad[ref_pix[high_diffs]] = True
                 int_bad[pair_pixel[high_diffs]] = True
                 int_bad[ref_pix[high_means]] = True
                 int_bad[pair_pixel[high_means]] = True
                 int_bad[ref_pix[high_stds]] = True
                 int_bad[pair_pixel[high_stds]] = True
 
-                log.debug(f'{np.sum(int_bad[offset:offset + amplifier])} '
-                          f'suspicious bad reference pixels in '
-                          f'amplifier {k}, integration {j}')
+                log.debug(
+                    f"{np.sum(int_bad[offset : offset + amplifier])} "
+                    f"suspicious bad reference pixels in "
+                    f"amplifier {k}, integration {j}"
+                )
                 mask_bad |= int_bad
 
         # replace any flagged pixels if desired
         if not flag_only:
             # list of all bad pixels
-            all_bad = np.arange(offset, offset + amplifier)[
-                mask_bad[offset:offset + amplifier]]
+            all_bad = np.arange(offset, offset + amplifier)[mask_bad[offset : offset + amplifier]]
             for bad_pix in all_bad:
-                replace_refpix(bad_pix, data, mask_bad, is_irs2,
-                               offset, offset + amplifier,
-                               scipix_n, refpix_r)
+                replace_refpix(
+                    bad_pix, data, mask_bad, is_irs2, offset, offset + amplifier, scipix_n, refpix_r
+                )
 
     if flag_only:
-        log.info(f'Total bad reference pixels flagged: {np.sum(mask_bad)}')
+        log.info(f"Total bad reference pixels flagged: {np.sum(mask_bad)}")
     else:
-        log.info(f'Total bad reference pixels replaced: {np.sum(mask_bad)}')
+        log.info(f"Total bad reference pixels replaced: {np.sum(mask_bad)}")
     if pixeldq is not None:
-        pixeldq[mask_bad] |= (dqflags.pixel['BAD_REF_PIXEL']
-                              | dqflags.pixel['DO_NOT_USE'])
+        pixeldq[mask_bad] |= dqflags.pixel["BAD_REF_PIXEL"] | dqflags.pixel["DO_NOT_USE"]
 
 
-def subtract_reference(data0, alpha, beta, irs2_mask, scipix_n,
-                       refpix_r, pad, preserve_refpix=False):
-    """Subtract reference output and pixels for the current integration.
+def subtract_reference(
+    data0, alpha, beta, irs2_mask, scipix_n, refpix_r, pad, preserve_refpix=False
+):
+    """
+    Subtract reference output and pixels for the current integration.
 
     Parameters
     ----------
-    data0: ramp data
+    data0 : ramp data
```

Comment:
what is the datatype specifically? is it a `JWSTRampModel` or an ndarray?

Reply:
ndarray